### PR TITLE
Added hasCookie and getCookie methods. Added tests for both hasCookie…

### PR DESCRIPTION
**Pull request by SaidShah**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

#32 Add HttpRequestExtensions.GetCookie method. I also added a HasCookie method.

### CODE REVIEW

File: src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs [hunk 1]
```diff
@@ -19,6 +19,34 @@ public static class HttpRequestExtensions
...
+
+        /// <summary>
+        /// Returns whether or not the specified cookie is found in the request
+        ///
+        /// If the cookie is found but its value is null/whitespace, it will return false.
+        /// </summary>
+        /// <param name="request">The request to check for the cookie</param>
+        /// <param name="name">The name of the cookie to check for</param>
+        /// <returns></returns>
+        public static bool HasCookie(this HttpRequest request, string name)
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs:40]: Nitpick - alphabetization of methods within a region (`HasCookie` should be after `GetIpAddress`)

File: test/AndcultureCode.CSharp.Extensions.Tests/HttpRequestExtensionsTest.cs [hunk 2]
```diff
@@ -20,6 +21,139 @@ public HttpRequestExtensionsTest(ITestOutputHelper output) : base(output)
...
+            Mock.Get(cookieCollection).Setup((e) => e.ContainsKey(key)).Returns(true);
+            Mock.Get(cookieCollection).SetupGet((e) => e[key]).Returns(value);
+            var sut = Mock.Of<HttpRequest>();
+            Mock.Get(sut).Setup((e) => e.Cookies).Returns(cookieCollection);
+
+            // Act && Assert
+            sut.HasCookie(key).ShouldBeTrue();
+        }
+
+        #endregion HasCookie
```
Comment (brandongregoryscott) [inline test/AndcultureCode.CSharp.Extensions.Tests/HttpRequestExtensionsTest.cs:155]: Nitpick - try to keep regions for tests alphabetized along with where they would be in the implementation

**Review by brandongregoryscott — APPROVED** [review]
Just some minor alphabetization to clean up 👍 Thanks for porting these, @SaidShah!

**Comment by SaidShah** [comment]
@brandongregoryscott - I have alphabetized the methods and regions. 👍

File: src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs [hunk 3]
```diff
@@ -48,6 +58,24 @@ public static string GetHeader(this HttpRequest request, string name)
...
+        {
+            var cookieCollection = request?.Cookies;
+            if (cookieCollection == null || !cookieCollection.ContainsKey(name) || string.IsNullOrWhiteSpace(cookieCollection[name]))
```
Comment (jstevenson) [inline src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs:72]: You should add a null check for "name" also

Comment (jstevenson) [reply src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs:72]: Also, add a test for a null "name" value.

**Review by jstevenson — APPROVED** [review]

File: src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs (hunk 3 above)
Comment (SaidShah) [reply src/AndcultureCode.CSharp.Extensions/HttpRequestExtensions.cs:72]: @jstevenson null names should be accounted for in the `!cookieCollection.ContainsKey(name)` so I don't think there is a need to add an extra condition in the if statement. But, I did add a test to confirm that null names were being handled. 👍

**Review by brandongregoryscott — APPROVED** [review]
